src/WinRT.Runtime/ExceptionHelpers.cs

```diff
@@ -51,133 +51,141 @@ static unsafe class ExceptionHelpers
-                        if (languageException is object)
-                        {
-                            if (languageException.IsReferenceToManagedObject)
-                            {
-                                ex = ComWrappersSupport.FindObject<Exception>(languageException.ThisPtr);
-                                if (GetHRForException(ex) == hr)
-                                {
-                                    restoredExceptionFromGlobalState = true;
-                                    return ex;
-                                }
-                            }
-                            else
-                            {
-                                // This could also be a proxy to a managed exception.
-                                hasOtherLanguageException = true;
-                            }
-                        }
-                    }
-
-                    if (hr == hrLocal)
+        [DllImport("oleaut32.dll")]
+        private static extern int SetErrorInfo(uint dwReserved, IntPtr perrinfo);
+
+        private static delegate* unmanaged[Stdcall]<IntPtr*, int> getRestrictedErrorInfo;
+        private static delegate* unmanaged[Stdcall]<IntPtr, int> setRestrictedErrorInfo;
+        private static delegate* unmanaged[Stdcall]<int, IntPtr, IntPtr, int> roOriginateLanguageException;
+        private static delegate* unmanaged[Stdcall]<IntPtr, int> roReportUnhandledError;
+
+        private static readonly bool initialized = Initialize();
+
+        private static bool Initialize()
+        {
+            IntPtr winRTErrorModule = Platform.LoadLibraryExW("api-ms-win-core-winrt-error-l1-1-1.dll", IntPtr.Zero, (uint)DllImportSearchPath.System32);
+            if (winRTErrorModule != IntPtr.Zero)
+            {
+#if NET7_0_OR_GREATER || CsWinRT_LANG_11_FEATURES
+                ReadOnlySpan<byte> langExceptionString = "RoOriginateLanguageException"u8;
+                ReadOnlySpan<byte> reportUnhandledErrorString = "RoReportUnhandledError"u8;
+#else
+                string langExceptionString = "RoOriginateLanguageException";
+                string reportUnhandledErrorString = "RoReportUnhandledError";
+#endif
+
+                roOriginateLanguageException = (delegate* unmanaged[Stdcall]<int, IntPtr, IntPtr, int>)Platform.GetProcAddress(winRTErrorModule, langExceptionString);
+                roReportUnhandledError = (delegate* unmanaged[Stdcall]<IntPtr, int>)Platform.GetProcAddress(winRTErrorModule, reportUnhandledErrorString);
+            }
+            else
+            {
+                winRTErrorModule = Platform.LoadLibraryExW("api-ms-win-core-winrt-error-l1-1-0.dll", IntPtr.Zero, (uint)DllImportSearchPath.System32);
+            }
+
+            if (winRTErrorModule != IntPtr.Zero)
+            {
+                getRestrictedErrorInfo = (delegate* unmanaged[Stdcall]<IntPtr*, int>)Platform.GetProcAddress(winRTErrorModule, "GetRestrictedErrorInfo"u8);

```

Comment:
ifdef here too?

Reply:
Not sure why it didn't show up in VS Code Search 😅. Will fix that.